ethereum/contracts/pyth/PythGovernance.sol

```diff
@@ -45,14 +45,17 @@ abstract contract PythGovernance is PythGetters, PythSetters, PythGovernanceInst
         if (gi.action == GovernanceAction.UpgradeContract) {
             require(gi.targetChainId != 0, "upgrade with chain id 0 is not possible");
             upgradeContract(gi.payload);
-        } else if (gi.action == GovernanceAction.SetGovernanceDataSource) {
-            setGovernanceDataSource(gi.payload);
+        } else if (gi.action == GovernanceAction.TransferGovernanceDataSource) {
+            require(gi.targetChainId == 0, "governance data source change should be applied to all chains");
```

Comment:
This seems like an unnecessary restriction, I'm not sure what this buys us.

Reply:
Based on my understanding the goal of governance is the be restricted as much as possible.

IMO it does not make sense that this action gets applied to only one single chain, and it has overhead of reverting it back. If you see any possible use case that it can happen then I will remove it, otherwise I think it should be there. It's not a security issue but it's avoiding possible mistakes and enforcing the behavior of this action.

Reply:
i think there are reasonable cases where we may want to shard governance authority to multiple emitters. e.g., we could make a council that owns the EVM contracts but not the Aptos contracts. I think we

Reply:
Ah got it, never thought of such a case. Ok then. I'll fix it.